src/app/services/wall.service.ts

```diff
@@ -0,0 +1,59 @@
+import { Injectable } from '@angular/core';
+import { Snake } from '../models/snake.model';
+import { Grid } from '../models/grid.model';
+import { Wall } from '../models/wall.model';
+import { RandomHelper } from '../helpers/random.helper';
+import { MAX_WALLS_COUNT, MIN_WALLS_COUNT, WALL_TEMPLATES } from '../constants/wall.constants';
+import { CellPosition } from '../models/cell-position.model';
+import { Cell } from '../models/cell.model';
+import { Food } from '../models/food.model';
+
+@Injectable({
+    providedIn: 'root',
+})
+export class WallService {
+    constructor(private randomHelper: RandomHelper) {}
+
+    generateWalls(grid: Grid, snake: Snake, food: Food): Wall[] {
+        const wallsCount = this.randomHelper.randomNumber(MIN_WALLS_COUNT, MAX_WALLS_COUNT);
+
+        let occupiedCells = snake.body.concat(food.cell);
+
+        let walls: Wall[] = [];
```

Comment:
Why let and not const?

Reply:
For some reason I was thinking they would change at some point like the apples haha